lib/nerves_hub_core/api.ex

```diff
@@ -137,4 +137,32 @@ defmodule NervesHubCore.API do
     |> Enum.map(&Certificate.from_pem!/1)
     |> Enum.map(&Certificate.to_der/1)
   end
+
+  defp get_api_host(opts) do
+    System.get_env("NERVES_HUB_HOST") || opts[:api_host]
+  end
+
+  defp get_api_port(opts) do
+    case System.get_env("NERVES_HUB_PORT") || opts[:api_port] do
+      port when is_binary(port) ->
+        try do
+          String.to_integer(port)
+        rescue
+          _e -> raise_port_error(port)
+        end
+
+      port when is_integer(port) ->
+        port
+
+      port ->
+        raise_port_error(port)
+    end
+  end
+
+  defp raise_port_error(port) do
+    raise """
+      NervesHub API port expects an integer
+      but is set to: #{port}
+    """
+  end
```

Comment:
The number of lines of code in that makes me sad. Would this be better (at the expense of uglier error messages, but we don't have pretty error messages for hostnames)?

```elixir
defp get_env_as_integer(str) do
  case System.get_env(str) do
    nil -> nil
    value -> String.to_integer(value)
  end
end

Then `get_env_as_integer("NERVES_HUB_PORT") || opts[:api_port]`